### DISCUSSION TITLE
v0.5 - The Update for Win11

### DISCUSSION THREAD

**Discussion by U-C-S** [U-C-S/Hurl#29] [Announcements]
## Feature Updates
- Update Browser Selection Window UI
  - Acrylic Background
  - Drop Shadows (Win 11 only)
  - Native Rounded Corners (Squary corners in Win10)
  - Border Color
  - Buttons and they use fluent icons
  - Settings btn which opens the `UserSettings.json` file in notepad
- Selection window expands automatically instead of a fixed length that previously only supported a view of 4 browsers
- Deprecate Hurl Settings (Will be remade with new UI in next updates)
- `UserSettings.json`
  - Remove unused setting fields used
  - Support for Launch Args - `LaunchArgs`
  - Support for hiding browsers - `hidden`
  - Support for `AlternateLaunches` (see [docs](https://github.com/U-C-S/Hurl/wiki/Editing-UserSettings.json#alternatelaunches))

## Development Updates
- Update to .NET 6

## What's Changed [AUTO-GENERATED]
* Backport dev to dotnet 6 branch by @U-C-S in https://github.com/U-C-S/Hurl/pull/16
* Add WPFUI and Acrylic Background  by @U-C-S in https://github.com/U-C-S/Hurl/pull/22
* Updates for #22 by @U-C-S in https://github.com/U-C-S/Hurl/pull/23
* v0.5 - Acrylic Background, WPFUI controls...... by @U-C-S in https://github.com/U-C-S/Hurl/pull/28

**Full Changelog**: https://github.com/U-C-S/Hurl/compare/v0.3.0...v0.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/U-C-S/Hurl/releases/tag/v0.5.0'>v0.5 - The Update for Win11</a>.</em>

**Comment by nitin2953** [comment]
Is `"AlternateLaunches"` in `UserSettings` working ??
I configured it properly but it is launching browser with new profile instead of specified profile argument

**Reply by U-C-S** [reply]
Yeh, It's perfectly working. I use it regularly. 

> I configured it properly but it is launching browser with new profile instead of specified profile argument

This behavior is not from Hurl, it comes from Browsers. Check whether you gave a correct Launch Arguments

Note: There's a "Default" profile, then any other profile created are named as "Profile 1", "Profile 2".....

**Reply by U-C-S** [reply]
Also, Check out the Updated Wiki: https://github.com/U-C-S/Hurl/wiki/Editing-UserSettings.json

**Reply by nitin2953** [reply]
Incognito is working BUT other profiles don't
I think it's a problem with my windows installation (it is heavily customized & debloated)

BTW if you have some time & BrowserSelect installed, try opening a `*.url` file with BrowserSelect whose name contains spaces.
If it **don't** open new tabs for every word after space in file name OR opening actual file itself `file:///*`
Then I have corrupted my system accidentally

**Reply by U-C-S** [reply]
> Incognito is working BUT other profiles don't

Can you post your `UserSettings.json` here ??

> If it don't open new tabs for every word after space in file name OR opening actual file itself file:///*

This is not a problem with your system. This is completely normal behavior and its just properly not handled by BrowserSelect. if you understand the command line args, then you might figure the problem here. BrowserSelect or Hurl just passes whatever arguments it receives directly to Browsers. 

like there's a difference b/w following 2 commands

chrome.exe "file://c/lol lol lol.txt"

chrome.exe file://c/lol lol lol.txt

The space are treated as separators in command-line args

**Reply by nitin2953** [reply]
> chrome.exe "file://c/lol lol lol.txt"

realize this now
But

**Reply by nitin2953** [reply]
<details>
<summary>UserSettings.json</summary>
<pre>
{
  "AppPath": "C:\\Users\\nitin\\AppData\\Roaming\\Hurl\\UserSettings.json",
  "Browsers": [
    {
      "SourceType": 0,
      "Name": "Firefox Developer Edition",
      "ExePath": "C:\\Program Files\\Firefox Developer Edition\\firefox.exe",
      "Hidden": false
    },
    {
      "SourceType": 0,
      "Name": "Google Chrome",
      "ExePath": "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe",
      "Hidden": false,
      "AlternateLaunches": [
        {
          "ItemName": "Main Profile",
          "LaunchArgs": "--profile-directory='Default'"
        },
        {
          "ItemName": "Profile 2",
          "LaunchArgs": "--profile-directory='Profile 1'"
        },
        {
          "ItemName": "Incognito",
          "LaunchArgs": "-incognito"
        }
      ]
    },
    {
      "SourceType": 0,
      "Name": "Google Chrome Beta",
      "ExePath": "C:\\Program Files\\Google\\Chrome Beta\\Application\\chrome.exe",
      "Hidden": false
    },
    {
      "SourceType": 0,
      "Name": "Google Chrome Dev",
      "ExePath": "C:\\Program Files\\Google\\Chrome Dev\\Application\\chrome.exe",
      "Hidden": false
    },
    {
      "SourceType": 0,
      "Name": "Microsoft Edge",
      "ExePath": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe",
      "Hidden": false
    }
  ],
  "LastUpdated": "13-Mar-22 6:44:45 PM"
}
</pre>
</details>
isn't the file is valid ?

**Reply by U-C-S** [reply]
YES... I'm getting your problem. I'll fix it....

**Reply by U-C-S** [reply]
Use escape characters... It's just in my UserSetting.json, I'm using escape characters `\` instead of `'` as used in wiki. I fixed it in wiki now. Thanks once again.

i.e. you need to use
```json
"LaunchArgs": "--profile-directory=\"Profile 1\""
```
instead of 
```json
"LaunchArgs": "--profile-directory='Profile 1'"
```

**Reply by nitin2953** [reply]
Profile value unquoted working, but other profiles can't be unquoted
---
```json
"LaunchArgs": "--profile-directory=Default"

"LaunchArgs": "--profile-directory='Profile 1'" // VALID BUT NOT WORKING

"LaunchArgs": "--profile-directory=Profile 1" // NOT VALID WON'T WORK
```

**Reply by nitin2953** [reply]
> i.e. you need to use
> 
> ```json
> "LaunchArgs": "--profile-directory=\"Profile 1\""
> ```

**WORKING FINE**

**Reply by eromrab** [reply]
> > i.e. you need to use
> > ```json
> > "LaunchArgs": "--profile-directory=\"Profile 1\""
> > ```
> 
> **WORKING FINE**

Sorry, I'm new here and only a very very basic developer type... is this saying you can set it up to have a button for multiple profiles?

I use Brave as my main, but I have a 3 different profiles on Chrome, so I'd love to  have a button for each profile of Chrome... 

Right now what I'm having to do is click on the browser profile I want it to open in, then click the link, and hit Chrome button, and it'll open in the last profile "touched."
